spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureBaseInterceptor.java

```diff
@@ -9,11 +9,14 @@
   protected final Class<? extends Throwable>[] expectedExceptions;
   protected final String reason;
   protected final String annotationUsed;
+  protected final boolean secondary;
 
-  public PendingFeatureBaseInterceptor(Class<? extends Throwable>[] expectedExceptions, String reason, String annotationUsed) {
+  public PendingFeatureBaseInterceptor(Class<? extends Throwable>[] expectedExceptions, String reason,
+                                       String annotationUsed, boolean secondary) {
     this.expectedExceptions = expectedExceptions;
     this.reason = reason;
     this.annotationUsed = annotationUsed;
+    this.secondary = secondary;
```

Comment:
please use a more expressive name

Reply:
I swapped the meaning and called it `failIfSuccessful` now.

Reply:
Replaced with custom exception